[#277] Use max voters instead of max votes (#278)
tqtezos/baseDAO: docs/specification.md

  ; max_proposals : nat
  // ^ Determine the maximum number of ongoing proposals that are allowed in the contract.
  ; max_voters : nat

>   // ^ Determine the maximum number of voters that are allowed to vote in the context of a proposal.

Reviewer: ```suggestion
  // ^ Determine the maximum number of voters that are allowed to vote on a proposal.
```
Nitpick, but this seems simpler.

Reviewer: Also I think that here it's worth adding a note that specifies that voters that votes on the same proposal in both ways are counted twice, what do you think?

Author: Oh, I think that is restriction will appear to be a bit arbitrary. Can we count them only once? It would make `vote` operation more costly. May be we can limit the voter count to `2 * max_voters`, allowing every voter to vote both ways in they choose.

What do you think?

Author: Right now I have modifed the spec to say the voters are counted twice. Will change it if you think otherwise.

| `QUORUM_NOT_MET`                     | A proposal is flushed, but there are not enough votes                                                       |
| `VOTING_STAGE_OVER`                  | Throws when trying to vote on a proposal that is already ended                                              |
| `MAX_PROPOSALS_REACHED`              | Throws when trying to propose a proposal when proposals max amount is already reached                       |

> | `MAX_VOTERS_REACHED`                 | Throws when trying to vote on a proposal when the votes max amount of that proposal is already reached      |

Reviewer: The description was not updated

Author: Have updated.

  from past stages that is not staked is less than specified `vote_amount` .
- Fails with `PROPOSAL_NOT_EXIST` if the proposal key is not associated with any ongoing proposals.
- Fails with `VOTING_STAGE_OVER` if the voting `stage` for the proposal has already ended.

> - Fails with `MAX_VOTERS_REACHED` if the amount of votes of the associated proposal

Reviewer: ```suggestion
- Fails with `MAX_VOTERS_REACHED` if the amount of voters of the associated proposal
```
